react/features/base/react/components/native/AvatarListItem.js

```diff
@@ -70,6 +70,44 @@ export default class AvatarListItem extends Component<Props> {
         this._renderItemLine = this._renderItemLine.bind(this);
     }
 
+    /**
+     * Helper function to render the content in the avatar container.
+     *
+     * @returns {React$Element}
+     */
+    _getAvatarContent() {
+        const {
+            avatarSize = AVATAR_SIZE,
+            avatarTextStyle
+        } = this.props;
+        const { avatar, title } = this.props.item;
+        const isAvatarURL = Boolean(avatar && avatar.match(/^http[s]*:\/\//i));
+
+        if (isAvatarURL) {
+            return (
+                <Avatar
+                    size = { avatarSize }
+                    uri = { avatar } />
+            );
+        }
+
+        if (avatar && !isAvatarURL) {
+            return (
+                <Icon name = { avatar } />
+            );
+        }
+
+        return (
+            <Text
+                style = { [
+                    styles.avatarContent,
+                    avatarTextStyle
+                ] }>
+                { title.substr(0, 1).toUpperCase() }
+            </Text>
+        );
+    }
```

Comment:
A pleasant surprise of a change.

Reply:
Flow continuously complained it didn't know which branch to take, and I don't blame it because I didn't know either :-)